### ISSUE TITLE
Bug on analyzer CC0031 (UseInvokeMethodToFireEvent) on parameterized functor

### ISSUE THREAD

**Issue by monoman** [code-cracker/code-cracker#397]
Needed to disable CC0031 on this code, because this fix can't be applied as ```TReturn``` is not constrained to a reference type so ```getter?.Invoke(result)``` would cause a syntax error.

``` csharp
public static TReturn GetAssemblyAttributeValue<T, TReturn>(
    this Assembly assembly, 
    Func<T, TReturn> getter) where T : Attribute
{
  var result = assembly.GetAssemblyAttribute<T>();
#pragma warning disable CC0031 // Use Invoke Method To Fire Event Analyzer
  return (result != null) ? getter(result) : default(TReturn);
#pragma warning restore CC0031 // Use Invoke Method To Fire Event Analyzer
}
```

Probably you need to obtain an ```ITypeParameterSymbol``` from the  return type on ```typeInfo``` and see if it has the ```HasReferenceTypeConstraint``` property set, if not it should return;

:cherries:

**Comment by giggio** [comment]
Bug confirmed. @monoman sorry, I don't know how I missed this.

**Comment by giggio** [comment]
Closed on #475.
